hummingbot/market/bitcoin_com/bitcoin_com_order_book_tracker.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python
+import asyncio
+import logging
+import hummingbot.market.bitcoin_com.bitcoin_com_constants as constants
+# import time
+
+from collections import defaultdict, deque
+from typing import Optional, Dict, List, Deque
+# from typing import Optional, Dict, List, Deque, Set
+from hummingbot.core.data_type.order_book_message import BitcoinComOrderBookMessage
+from hummingbot.logger import HummingbotLogger
+from hummingbot.core.data_type.order_book_tracker import OrderBookTracker, OrderBookTrackerDataSourceType
+from hummingbot.core.data_type.order_book_tracker_data_source import OrderBookTrackerDataSource
+# from hummingbot.core.data_type.order_book_tracker_entry import BitcoinComOrderBookTrackerEntry
+from hummingbot.core.utils.async_utils import safe_ensure_future
+from hummingbot.market.bitcoin_com.bitcoin_com_active_order_tracker import BitcoinComActiveOrderTracker
+from hummingbot.market.bitcoin_com.bitcoin_com_api_order_book_data_source import BitcoinComAPIOrderBookDataSource
+from hummingbot.market.bitcoin_com.bitcoin_com_order_book import BitcoinComOrderBook
+
+
+class BitcoinComOrderBookTracker(OrderBookTracker):
```

Comment:
@nionis As discussed earlier, you would need to revisit the `BitcoinComOrderBookTracker`. From my initial findings, the diff messages are not being applied to the order book.

![h3tfFEVkSs6Wh4QKMPPPBA_mini_1](https://user-images.githubusercontent.com/6755922/68343112-f0075080-00a0-11ea-8279-1493e42e9422.jpg)

Looking at the `last_diff_uid` in the `OrderBook`, it is not being updated. This should not happen if the diff messages have been successfully applied.

Reply:
With the current implementation you are using the `_track_single_book` function in the `OrderBookTracker` base class and the message(s) queued are not parsed as expected. As such, it is throwing an exception when `apply_diffs` is being called.

Reply:
I have addressed this issue in this [commit](https://github.com/CoinAlpha/hummingbot/pull/1015/commits/731fc8cd41a27204c027dbdb9a14a056788311a8) and added a simple `last_diff_uid` check in the current [test](https://github.com/CoinAlpha/hummingbot/pull/1015/commits/731fc8cd41a27204c027dbdb9a14a056788311a8#diff-4b5a3fde1bc6491d5a18e140db1ea07dR94).